contracts/proxy/VolatilePoolProxy.sol

```diff
@@ -52,90 +59,140 @@ contract VolatilePoolProxy is Initializable, OwnableUpgradeSafe, Helper {
+        @param advisorVolatilePoolToken address of advisors volatile pool token.
+        @param volatileProtocolStableCoinProportion stable coin proportion in volatile pool.
+        @param volatileProtocolVolatileCoinProportion volatile coin proportion in volatile pool.
+        // Reusing some vars due to stack too deep error
+     */
     function redeemAmount(
         address _investor,
         address _advisor,
-        address[] calldata _vault,
         address _stablecoin,
-        uint256 _investorStablePoolLiquidity,
         uint256 _investorVolatilePoolLiquidity,
-        address advisorStablePoolToken,
         address advisorVolatilePoolToken,
-        uint256 yearnAdvisorProportion
-    ) external {
-        for (uint256 i = 0; i < _vault.length; i++) {
-            // had to reduce vars due to sol stack too deep error
-            if (_vault[i] == getETHVault()) {
-                uint256 poolTokenPrice = YearnVault(_vault[i]).balanceOf(address(this)).div(IERC20(advisorVolatilePoolToken).totalSupply());
-                uint256 investorRedeemAmount = _investorVolatilePoolLiquidity.mul(poolTokenPrice);
-                YearnVault(_vault[i]).withdrawETH(investorRedeemAmount);
-                uint256 investorReturns = getInvestorReturnAmount(_vault[i], investorRedeemAmount);
-                // 1 % per investor
-                uint256 advisorFee = (investorReturns.mul(1)).div(100);
-                investorReturns = investorReturns.sub(advisorFee);
-                (bool ethTransferCheck, ) = _advisor.call{
-            value: advisorFee
-            }("");
-                require(ethTransferCheck, "Advisor Transfer failed.");
-                (ethTransferCheck, ) = _investor.call{
-            value: investorReturns
-            }("");
-                require(ethTransferCheck, "Investor Transfer failed.");
-                // transfer eth to both
-            } else {
-                uint256 investorYearnLiquidity = _investorStablePoolLiquidity.mul(yearnAdvisorProportion).div(100);
-                uint256 investorRedeemAmount = investorYearnLiquidity.mul(YearnVault(_vault[i]).balanceOf(address(this)).div(IERC20(advisorStablePoolToken).totalSupply()));
-                YearnVault(_vault[i]).withdraw(investorRedeemAmount);
-                uint256 investorReturns = getInvestorReturnAmount(_vault[i], investorRedeemAmount);
-                // 1 % per investor
-                uint256 advisorFee = (investorReturns.mul(1)).div(100);
-                investorReturns = investorReturns.sub(advisorFee);
-                IERC20(_stablecoin).safeTransfer(_advisor, advisorFee);
-                IERC20(_stablecoin).safeTransfer(
-                    _investor,
-                    investorReturns
-                );
-            }
+        uint256 volatileProtocolStableCoinProportion,
+        uint256 volatileProtocolVolatileCoinProportion
+    ) external returns (uint256) {
+        uint256 stableAssetFees = 0;
```

Comment:
We should really get some tests on these pools, there's a lot here that could go wrong